src/drc/api/viewsets.py

```diff
@@ -114,6 +119,50 @@ class EnkelvoudigInformatieObjectViewSet(NotificationViewSetMixin,
     notifications_kanaal = KANAAL_DOCUMENTEN
     audit = AUDIT_DRC
 
+    def get_queryset(self):
+        """
+        Retrieve the latest version of each EnkelvoudigInformatieObject
+        """
+        qs = super().get_queryset()
+        return qs.order_by('canonical', '-versie').distinct('canonical')
+
+    def get_object(self):
+        if 'versie' in self.request.query_params:
+            qs = super().get_queryset()
+            obj = get_object_or_404(
+                qs,
+                uuid=self.kwargs['uuid'],
+                versie=self.request.query_params['versie'],
+            )
+        elif 'registratieOp' in self.request.query_params:
```

Comment:
I feel this should be implemented in the `EnkelvoudigInformatieObjectFilterSet` instead. Especially since this query param is not picked up by schema generation now.